src/probability/martingale/convergence.lean

```diff
@@ -231,4 +251,242 @@ mem_ℒp_limit_process_of_snorm_bdd
+  have hgen : (⨆ n, ℱ n) = measurable_space.generate_from {s | ∃ n, measurable_set[ℱ n] s},
+  { ext s,
+    rw measurable_space.measurable_set_Sup,
+    simp_rw set.mem_range,
+    change _ ↔ measurable_space.generate_measurable {s | ∃ n, measurable_set[ℱ n] s} s,
+    simp only [exists_prop, exists_exists_eq_and] },
+  refine @measurable_space.induction_on_inter _ _ _ (⨆ n, ℱ n) hgen _ _ _ _ _ _ hs,
+  { rintro s ⟨n, hs⟩ t ⟨m, ht⟩ -,
+    by_cases hnm : n ≤ m,
+    { exact ⟨m, (ℱ.mono hnm _ hs).inter ht⟩ },
+    { exact ⟨n, hs.inter (ℱ.mono (not_le.1 hnm).le _ ht)⟩ } },
+  { simp only [measure_empty, with_top.zero_lt_top, measure.restrict_empty,
+      integral_zero_measure, forall_true_left] },
+  { rintro t ⟨n, ht⟩ -,
+    exact this n _ ht },
+  { rintro t htmeas ht -,
+    have hgeq := @integral_add_compl _ _ (⨆ n, ℱ n) _ _ _ _ _ _ htmeas (hintg.trim hle hgmeas),
+    have hheq := @integral_add_compl _ _ (⨆ n, ℱ n) _ _ _ _ _ _ htmeas
+      (hlimint.trim hle strongly_measurable_limit_process),
+    rw [add_comm, ← eq_sub_iff_add_eq] at hgeq hheq,
+    rw [set_integral_trim hle hgmeas htmeas.compl,
+      set_integral_trim hle strongly_measurable_limit_process htmeas.compl,
+      hgeq, hheq, ← set_integral_trim hle hgmeas htmeas,
+      ← set_integral_trim hle strongly_measurable_limit_process htmeas,
+      ← integral_trim hle hgmeas, ← integral_trim hle strongly_measurable_limit_process,
+      ← integral_univ, this 0 _ measurable_set.univ, integral_univ, ht (measure_lt_top _ _)] },
+  { rintro f hf hfmeas heq -,
+    rw [integral_Union (λ n, hle _ (hfmeas n)) hf hintg.integrable_on,
+      integral_Union (λ n, hle _ (hfmeas n)) hf hlimint.integrable_on],
+    exact tsum_congr (λ n, heq _ (measure_lt_top _ _)) }
+end
+
+/-- Part c of the **L¹ martingale convergnce theorem**: Given a integrable function `g` which
+is measurable with respect to `⨆ n, ℱ n` where `ℱ` is a filtration, the martingale defined by
+`𝔼[g | ℱ n]` converges in L¹ to `g`.
+
+This martingale also converges to `g` almost everywhere and this result is provided by
+`measure_theory.mem_ℒp.tendsto_ae_condexp` -/
+lemma mem_ℒp.tendsto_snorm_condexp
+  (hg : mem_ℒp g 1 μ) (hgmeas : strongly_measurable[⨆ n, ℱ n] g) :
+  tendsto (λ n, snorm (μ[g | ℱ n] - g) 1 μ) at_top (𝓝 0) :=
+tendsto_Lp_of_tendsto_in_measure _ le_rfl ennreal.one_ne_top
+  (λ n, (strongly_measurable_condexp.mono (ℱ.le n)).ae_strongly_measurable) hg
+  ((mem_ℒp_one_iff_integrable.1 hg).uniform_integrable_condexp_filtration).2.1
+    (tendsto_in_measure_of_tendsto_ae
+    (λ n,(strongly_measurable_condexp.mono (ℱ.le n)).ae_strongly_measurable)
+      (hg.tendsto_ae_condexp hgmeas))
+
+/-- **Lévy's upward theorem**, almost everywhere version: given a function `g` and a filtration
+`ℱ`, the sequence defined by `𝔼[g | ℱ n]` converges almost everywhere to `𝔼[g | ⨆ n, ℱ n]`. -/
+lemma mem_ℒp.tendsto_ae_condexp' (g : Ω → ℝ) :
```

Comment:
why is the name of the form `mem_ℒp.*`? There is no `mem_ℒp` hypothesis. Same question for the next lemma.